src/GraphQL/Telemetry/GraphQLTelemetryProvider.cs

```diff
@@ -62,6 +59,25 @@ public virtual async Task<ExecutionResult> ExecuteAsync(ExecutionOptions options
         if (activity == null)
             return await next(options).ConfigureAwait(false);
 
+        if (!activity.IsAllDataRequested)
+            return await next(options).ConfigureAwait(false);
+
+        try
+        {
+            if (!_telemetryOptions.Filter(options))
+            {
+                activity.IsAllDataRequested = false;
+                activity.ActivityTraceFlags &= ~ActivityTraceFlags.Recorded;
+                return await next(options).ConfigureAwait(false);
+            }
+        }
+        catch
+        {
+            activity.IsAllDataRequested = false;
+            activity.ActivityTraceFlags &= ~ActivityTraceFlags.Recorded;
+            return await next(options).ConfigureAwait(false);
+        }
```

Comment:
```suggestion
        if (!_telemetryOptions.Filter(options))
        {
            activity.IsAllDataRequested = false;
            activity.ActivityTraceFlags &= ~ActivityTraceFlags.Recorded;
            return await next(options).ConfigureAwait(false);
        }
 ```
I don't think we should catch exceptions within the filter delegate.  Also with the prior code, exceptions thrown from `next(options)` would trigger running `next(options)` again.

Reply:
That how it's done in other instrumentations. The philosophy of OTel says telemetry bugs, shouldn't break the application.

Reply:
You are right about the "trigger again". Will check it tommorow.

Reply:
These are not telemetry bugs.  This is user code.  Just as if there was a bug in any other user code.  Sure if you were OpenTelemetry writing a plugin, you wouldn't want auto-enabled telemetry in GraphQL to then crash the user's application.  But the filter delegate is user code.

Reply:
You are right, this is user code. But this code is related to telemetry and not to business logic. It's ok to fail to start the application if telemetry is misconfigured, but failing the already running application when emitting traces is less acceptable.
Literally every instrumentation in core and contrib OTel repos catching errors from the filter delegate, and users will expect the same from GraphQL:
[AspNet](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/6bb57e25b284ed500981a91afa6156c8ed0fa4d4/src/OpenTelemetry.Instrumentation.AspNet/Implementation/HttpInListener.cs#L52-L74)
[AspNetCore](https://github.com/open-telemetry/opentelemetry-dotnet/blob/0889e8dc32badbb4c9be8fd24a974fb7e61fa779/src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs#L155-L171)
[Http](https://github.com/open-telemetry/opentelemetry-dotnet/blob/0889e8dc32badbb4c9be8fd24a974fb7e61fa779/src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerDiagnosticListener.cs#L120-L136)
[EF Core](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/6bb57e25b284ed500981a91afa6156c8ed0fa4d4/src/OpenTelemetry.Instrumentation.EntityFrameworkCore/Implementation/EntityFrameworkDiagnosticListener.cs#L169-L189)
[SqlClient](https://github.com/open-telemetry/opentelemetry-dotnet/blob/0889e8dc32badbb4c9be8fd24a974fb7e61fa779/src/OpenTelemetry.Instrumentation.SqlClient/Implementation/SqlClientDiagnosticListener.cs#L76-L92)
[Hangfire](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/6bb57e25b284ed500981a91afa6156c8ed0fa4d4/src/OpenTelemetry.Instrumentation.Hangfire/Implementation/HangfireInstrumentationJobFilterAttribute.cs#L55-L69)

The only difference is they are not silently swallowing these exceptions but logging `EventSource` events. We can do the same if this PR is merged https://github.com/graphql-dotnet/graphql-dotnet/pull/3539.

I can add a note about this behavior in XML docs to make it clear. Let me know what is your final decision on this, and I'll change the code accordingly.

Reply:
Well, if we don't have any MS references to copy, then we will need to refer to OTel sources.  Maybe we should review #3539 first.

Reply:
It just seems like a critical design error to swallow exceptions.  I can't imagine why tracing would be disabled when an exception was encountered either.  Wouldn't that just make it harder to track down what went wrong?  I'm okay following a standardized pattern, but just saying...

Reply:
At least with #3539 then it gets logged....somewhere...we hope...

Reply:
To be honest, `EventSource` feels like very old technology that is only old .NET Framework developers are familiar with 😅
I'd prefer to use `ILogger` for internal logging. If you don't want to get dependent on `Microsoft.Extensions.Logging.Abstuctions` package, we can introduce an internal logging interface + adapter for `ILogger` similar to what you did with `IServiceRegister => IServiceCollection`

Reply:
My thoughts also https://github.com/graphql-dotnet/graphql-dotnet/pull/3539#issuecomment-1437030956

Reply:
What about removing the exception swallowing for now until we introduce some type of logging mechanism?

Reply:
I'd be open to adding some dependencies to GraphQL.NET for v8 - specifically:

- [Microsoft.Extensions.DependencyInjection.Abstractions](https://www.nuget.org/packages/Microsoft.Extensions.DependencyInjection.Abstractions)
- [System.Diagnostics.DiagnosticSource](https://www.nuget.org/packages/System.Diagnostics.DiagnosticSource)
- [Microsoft.Extensions.Logging.Abstractions](https://www.nuget.org/packages/Microsoft.Extensions.Logging.Abstractions)

Each of these have around 4 billion downloads.  They are not going anywhere and have minimal API exposure.  We can reference older versions to ensure that there are no versioning conflicts.

Reply:
> What about removing the exception swallowing for now until we introduce some type of logging mechanism?

Ok

Reply:
Some thoughts about the subject. TL;DR: use `ILogger`
https://github.com/dotnet/diagnostics/issues/1605